spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BrokerEventListener.java

```diff
@@ -0,0 +1,218 @@
+	}
+
+	/**
+	 * Construct an instance using the supplied listener container factory and event keys.
+	 * Event keys are patterns to match routing keys for events published to the
+	 * {@code amq.rabbitmq.event} topic exchange. They can therefore match wildcards;
+	 * examples are {@code user.#, queue.created}. Refer to the plugin documentation for
+	 * information about available events. The container's lifecyle will be not be
+	 * controlled by this object's {@link SmartLifecycle} methods. The container should
+	 * not be configured with queues or a {@link MessageListener}; those properties will
+	 * be replaced.
+	 * @param container the listener container.
+	 * @param eventKeys the event keys.
+	 */
+	public BrokerEventListener(AbstractMessageListenerContainer container, String... eventKeys) {
+		this(container, false, eventKeys);
+	}
+
+	private BrokerEventListener(AbstractMessageListenerContainer container, boolean ownContainer, String... eventKeys) {
+		Assert.notNull(container, "listener container cannot be null");
+		Assert.isTrue(!ObjectUtils.isEmpty(eventKeys), "At least one event key is required");
+		this.container = container;
+		this.container.setQueues(this.eventQueue);
+		this.container.setMessageListener(this);
+		this.eventKeys = Arrays.copyOf(eventKeys, eventKeys.length);
+		this.container.getConnectionFactory().addConnectionListener(this);
+		this.admin = new RabbitAdmin(this.container.getConnectionFactory());
+		this.ownContainer = ownContainer;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	/**
+	 * Return any exception thrown when attempting to bind the queue to the event exchange.
+	 * @return the exception.
+	 */
+	public @Nullable Exception getBindingsFailedException() {
+		return this.bindingsFailedException;
+	}
+
+	@Override
+	public synchronized void start() {
+		if (!this.running) {
+			if (this.stopInvoked) {
+				// redeclare auto-delete queue
+				this.stopInvoked = false;
+				onCreate(null);
+			}
+			if (this.ownContainer) {
+				this.container.start();
+			}
+			this.running = true;
+		}
+	}
+
+	@Override
+	public synchronized void stop() {
+		if (this.running) {
+			if (this.ownContainer) {
+				this.container.stop();
+			}
+			this.running = false;
+			this.stopInvoked = true;
+		}
+	}
+
+	@Override
+	public synchronized boolean isRunning() {
+		return this.running;
+	}
+
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	public void setPhase(int phase) {
+		this.phase = phase;
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return this.autoStartup;
+	}
+
+	public void setAutoStartup(boolean autoStartup) {
+		this.autoStartup = autoStartup;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		callback.run();
+	}
+
+	@Override
+	public void onMessage(Message message) {
+		if (this.applicationEventPublisher != null) {
```

Comment:
I think we need to add a WARN log in the `else`.
show the event we have received and mention the it would be better to add an `ApplicationEventPublisher`